### PR TITLE
feat: oss get metadata

### LINKED ISSUE #1664: [dfget] 希望支持多线程从oss回源
现在首次从oss回源有点慢，希望能支持多线程，回源速度更快
感谢~

### PR DESCRIPTION
**Pull request by jim3ma**
Signed-off-by: Jim Ma <majinjing3@gmail.com>

## Related Issue

Fix #1664

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: pkg/source/source_client.go [hunk 1]
```diff
@@ -389,9 +390,10 @@ func GetMetadata(request *Request) (*Metadata, error) {
 	if !ok {
 		return nil, fmt.Errorf("scheme %s: %w", request.URL.Scheme, ErrNoClientFound)
 	}
-	getter, ok := client.(*clientWrapper).rc.(ResourceMetadataGetter)
+	cw := client.(*clientWrapper)
+	getter, ok := cw.rc.(ResourceMetadataGetter)
```
Comment (bigerous) [inline pkg/source/source_client.go:394]: This not support using source client plugin. 这种写法不支持使用plugin模式的source client
参考https://github.com/dragonflyoss/Dragonfly2/blob/a9d4b1b07bfee9671c62e26c948de703d00b6ac9/pkg/source/source_client.go#L371

File: pkg/source/clients/ossprotocol/oss_source_client.go [hunk 2]
```diff
@@ -135,6 +135,50 @@ func (osc *ossSourceClient) IsSupportRange(request *source.Request) (bool, error
...
+		Header:             hdr,
+		Status:             http.StatusText(http.StatusOK),
+		StatusCode:         http.StatusOK,
+		SupportRange:       true,
+		TotalContentLength: totalContentLength,
+		Validate: func() error {
+			return nil
+		},
+		Temporary: func() bool {
+			return false
```
Comment (bigerous) [inline pkg/source/clients/ossprotocol/oss_source_client.go:177]: should return true

Comment (jim3ma) [reply pkg/source/clients/ossprotocol/oss_source_client.go:177]: The `source.Metadata` is valid, `Temporary ` will not be used.

**Review by bigerous — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**